pkg/controller/servicebindingrequest/reconciler_test.go

```diff
@@ -184,3 +184,43 @@ func TestReconcilerReconcileUsingVolumes(t *testing.T) {
 		require.Equal(t, reconcilerName, volumes[0].VolumeSource.Secret.SecretName)
 	})
 }
+
+//TestReconcilerReconcileWithConflictingAppSelc tests when sbr has conflicting ApplicationSel such as MatchLabels=App1 and ResourceRef=App2 it should prioritise the ResourceRef
+func TestReconcilerReconcileWithConflictingAppSelc(t *testing.T) {
+	backingServiceResourceRef := "backingServiceRef"
+	applicationResourceRef1 := "applicationResourceRef1"
+	matchLabels1 := map[string]string{
+		"connects-to": "database",
+		"environment": "testing",
+	}
+	applicationResourceRef2 := "applicationResourceRef2"
+
+	f := mocks.NewFake(t, reconcilerNs)
+
+	f.AddMockedUnstructuredDeployment(applicationResourceRef1, matchLabels1)
+	f.AddMockedUnstructuredDeployment(applicationResourceRef2, nil)
+	f.AddMockedUnstructuredServiceBindingRequest(reconcilerName, backingServiceResourceRef, applicationResourceRef2, deploymentsGVR, matchLabels1)
+	f.AddMockedUnstructuredCSV("cluster-service-version-list")
+	f.AddMockedUnstructuredDatabaseCRD()
+	f.AddMockedUnstructuredDatabaseCR(backingServiceResourceRef)
+	f.AddMockedSecret("db-credentials")
+
+	fakeClient := f.FakeClient()
+	fakeDynClient := f.FakeDynClient()
+	reconciler := &Reconciler{client: fakeClient, dynClient: fakeDynClient, scheme: f.S}
+
+	t.Run("test-reconciler-reconcile-with-conflicting-application-selector", func(t *testing.T) {
+
+		res, err := reconciler.Reconcile(reconcileRequest())
+		require.NoError(t, err)
+		require.False(t, res.Requeue)
+
+		namespacedName := types.NamespacedName{Namespace: reconcilerNs, Name: reconcilerName}
+		sbrOutput, err := reconciler.getServiceBindingRequest(namespacedName)
+		require.NoError(t, err)
+
+		require.Equal(t, BindingSuccess, sbrOutput.Status.BindingStatus)
+		expectedStatus := fmt.Sprintf("%s/%s", reconcilerNs, applicationResourceRef2)
+		require.Equal(t, expectedStatus, sbrOutput.Status.ApplicationObjects[0])
```

Comment:
Can you add a check that length of `sbrOutput.Status.ApplicationObjects` is 1 and there are no other applicationObjects?